tests/phpunit/tests/Stories_Renderer/Renderer.php

```diff
@@ -0,0 +1,358 @@
+
+		ob_start();
+		$this->call_private_method( $renderer, 'get_content_overlay', [ [] ] );
+		$output = ob_get_clean();
+
+		$this->assertEmpty( $output );
+
+		$story_data = [
+			'title'                => 'Story Title',
+			'date'                 => 'November 11, 2020',
+			'author'               => 'admin',
+			'show_content_overlay' => true,
+		];
+
+		ob_start();
+		$this->call_private_method( $renderer, 'get_content_overlay', [ $story_data ] );
+		$output = ob_get_clean();
+
+		$this->assertContains( 'By admin', $output );
+		$this->assertContains( 'On November 11, 2020', $output );
+		$this->assertContains( 'Story Title', $output );
+	}
+
+	/**
+	 * @covers ::get_container_styles
+	 */
+	public function test_get_container_styles() {
+
+		$this->stories->method( 'get_story_attributes' )->willReturn(
+			[
+				'view_type'         => 'grid',
+				'number_of_columns' => '3',
+			]
+		);
+
+		$renderer = new \Google\Web_Stories\Stories_Renderer\Generic_Renderer( $this->stories );
+
+		$expected = 'grid-template-columns:repeat(3, 1fr);';
+		$output   = $this->call_private_method( $renderer, 'get_container_styles' );
+
+		$this->assertEquals( $expected, $output );
+	}
+
+	/**
+	 * @covers ::get_single_story_classes
+	 */
+	public function test_get_single_story_classes() {
+
+		$this->stories->method( 'get_story_attributes' )->willReturn(
+			[
+				'show_story_poster' => true,
+				'view_type'         => 'circles',
+			]
+		);
+
+		$renderer = new \Google\Web_Stories\Stories_Renderer\Generic_Renderer( $this->stories );
+		$expected = 'web-stories-list__story-wrapper has-poster';
+
+		$output = $this->call_private_method( $renderer, 'get_single_story_classes' );
+
+		$this->assertEquals( $expected, $output );
+	}
+
+	/**
+	 * @covers ::get_container_classes
+	 */
+	public function test_get_container_classes() {
+
+		$this->stories->method( 'get_story_attributes' )->willReturn(
+			[
+				'view_type'         => 'circles',
+				'class'             => 'test',
+				'show_story_poster' => false,
+			]
+		);
+
+		$renderer = new \Google\Web_Stories\Stories_Renderer\Generic_Renderer( $this->stories );
+
+		$expected = 'web-stories-list is-view-type-circles alignnone test';
+
+		$output = $this->call_private_method( $renderer, 'get_container_classes' );
+
+		$this->assertEquals( $expected, $output );
+	}
+
+	/**
+	 * @covers ::maybe_render_archive_link
+	 */
+	public function test_maybe_render_archive_link() {
+
+		$this->stories->method( 'get_story_attributes' )->willReturn(
+			[
+				'show_story_poster'         => false,
+				'show_stories_archive_link' => true,
+				'stories_archive_label'     => 'View All Stories',
+			]
+		);
+
+		$renderer = new \Google\Web_Stories\Stories_Renderer\Generic_Renderer( $this->stories );
+
+		$archive_link = get_post_type_archive_link( \Google\Web_Stories\Story_Post_Type::POST_TYPE_SLUG );
+		ob_start();
+		$this->call_private_method( $renderer, 'maybe_render_archive_link' );
+		$expected = ob_get_clean();
```

Comment:
```suggestion
		$expected = get_echo( $this->call_private_method( $renderer, 'maybe_render_archive_link' ) );
```

Reply:
@spacedmonkey 
I think this would not be feasible because `call_private_method` is a protected method and `get_echo` cannot call it.

Also, the mentioned snipper is not correct, it should be 

```php
$expected = get_echo( array( $this, 'call_private_method' ), array( $renderer, 'maybe_render_archive_link' ) );
```